src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/crushmap/crushmap.component.ts

```diff
@@ -34,12 +34,24 @@ export class CrushmapComponent implements OnInit {
       };
     }
 
-    const rootNodeId = nodes[0].id || null;
+    const roots = [];
     nodes.reverse().forEach((node) => {
+      if (node.type === 'root') {
+        roots.push(node.id);
+      }
       treeNodeMap[node.id] = this.generateTreeLeaf(node, treeNodeMap);
     });
 
-    return treeNodeMap[rootNodeId];
+    const children = roots.map((id) => {
+      return treeNodeMap[id];
+    });
+
+    const result: TreeModel = {
+      value: 'CRUSH map',
+      children: children
+    };
+
+    return result;
```

Comment:
How about combining result and return?

```ts
    return {
      value: 'CRUSH map',
      children: children
    } as TreeModel;
```
Also I think you do not need to specify the model here as it includes the only needed attribute to be a valid `TreeModel` which is `value`.

Reply:
Done